flocker/node/agents/test/test_blockdevice.py

```diff
@@ -633,6 +640,48 @@ def test_ignore_deleted_datasets(self):
         self.assertEqual(expected_changes, actual_changes)
 
 
+class BlockDeviceDeployerResizeCalculateNecessaryStateChangesTests(
+        SynchronousTestCase
+):
+    """
+    Tests for ``BlockDeviceDeployer.calculate_necessary_state_changes``
+    in the cases relating to resizing a dataset.
+    """
+    def test_maximum_size_unchanged(self):
+        """
+        ``BlockDeviceDeployer.calculate_necessary_state_changes`` does not
+        calculate a change to shrink or grow datasets if the maximum_size of
+        the dataset has not changed.
+        """
```

Comment:
I left a comment in the new design about the need to supply the local_state in this case...in which case, perhaps this test can pass.